app/components/restoration_game.rb

```diff
@@ -0,0 +1,163 @@
+class RestorationGame < ViewComponent::Base
+  include Motion::Component
+  include RestoConstants
+  include RestorationInitialize
+  include WorldGeneration
+
+  attr_reader :board, :board_viewed, :index, :selected, :board_size, :view_size, :view_corner, :time_passed, :zoom
+  attr_reader :berries, :seeds, :saplings, :water, :water_used
+  attr_reader :saplings_to_give, :seeds_to_give
+  attr_reader :show_info, :show_intro_msg, :show_win_msg, :won
+
+  map_motion :change_selected
+  map_motion :close_intro_msg
+  map_motion :close_win_msg
+  map_motion :paint
+  map_motion :pan
+  map_motion :toggle_info_msg
+  every 1.second, :update
+
+  def initialize
+    start
+    generate_world
+  end
+
+  def change_selected(event)
+    @selected = event.target.data[:value].to_i
+  end
+
+  def check_win
+    if get_tiles(5).size == 0 && @won == 0
+      @won = 1
+      @show_win_msg = 1
+    end
+  end
+
+  def close_intro_msg
+    @show_intro_msg = 0
+  end
+
+  def close_win_msg
+    @show_win_msg = 0
+  end
+
+  def paint(event)
+    coord = event.target.data["coord"].to_i
+
+    place(coord)
+  end
+
+  def pan(event)
+    direction = event.target.data["value"].to_i
+    case direction
+    when RestoConstants::UP
+      @view_corner -= @board_size if @view_corner >= @board_size
+    when RestoConstants::DOWN
+      @view_corner += @board_size if @view_corner / @board_size < @board_size - @view_size
+    when RestoConstants::LEFT
+      @view_corner -= 1 if @view_corner % @board_size > 0
+    when RestoConstants::RIGHT
+      @view_corner +=1 if @view_corner % @board_size < @board_size - @view_size
+    end
+  end
+
+  def toggle_info_msg(event)
+    @show_info = event.target.data["value"].to_i
+  end
+
+  private
+
+  def evaluate(tile)
+    case @board[tile]
```

Comment:
Something I'd like to introduce to you here by way of example is the idea of using barewords instead of instance variables directly. The concept of barewords comes from Perl but is useful in Ruby as well. The distinction is this:

```ruby
case @board[tile] # instance variable

case board[tile] # bareword
```

Because you've got `attr_reader` at the top of this class, there is a method called `def board` that will return the instance variable `@board` for you. Wherever possible, you should prefer using barewords to using instance variables directly. Doing this allows you to change the source and scope of information, without altering the code that uses it.

A bareword can refer to a local variable, a method parameter, or a method call without any changes to the code that uses it. It's the most flexible way to use a variable.

There are some circumstances where you need to refer to the instance variable directly, namely, assignment.

```ruby
@num = 5 # @ is necessary for assignment

attr_accessor :meaning
self.meaning = 42 # no @ required, but explicit self IS required for a method call to `#other=`

@hsh = {}
hsh[:lexie] = :coolest # @ not necessary for setting a value in a hash though!
```

Outside of assignment though, you can refactor most references to instance vars to use barewords instead. I'd like you to back through this class and update anywhere you can to use barewords instead of `@vars`.

Further reading/video: https://www.rubytapas.com/2018/06/05/barewords/

Reply:
I think I understand what you're getting at here, thank you. You only need to directly reference the variable if it's not defined beforehand?